src/Symfony/Component/RateLimiter/Exception/LogicException.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter\Exception;
+
+/**
+ * Base LogicException for RateLimiter component.
+ *
+ * @author Valentin Silvestre <vsilvestre.pro@gmail.com>
+ */
+class LogicException extends \LogicException implements ExceptionInterface
```

Comment:
I think we can remove the custom exception classes/interfaces and use `\LogicException`. Logic exceptions are compile-time, they should be fixed and not handled (and thus custom classes aren't needed).